### PR TITLE
[omni,data] refactor: unify Qwen-Omni data processing into shared vlm_data_process

### PR DESCRIPTION
**Pull request by TimYangst**
### What does this PR do?

> Extracts the duplicated `process_sample` function from `train_qwen25_omni.py` and `train_qwen3_omni.py` into a shared `process_sample_qwen_omni` function in `tasks/data/vlm_data_process.py`. Introduces `get_omni_token_ids` to dynamically resolve image/video/audio pad token IDs from the processor vocab, enabling a single implementation to support both Qwen2.5-Omni and Qwen3-Omni-MoE without hardcoded token IDs.

### Checklist Before Starting

- [ ] Search for similar PRs. Paste at least one query link here: ...
- [ ] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `misc`, `ci`, `config`, `docs`, `core`, `data`, `dist`, `omni`, `logging`, `model`, `optim`, `ckpt`, `release`, `task`
  - If this PR involves multiple modules, separate them with `,` like `[megatron, torch, liger-kernals,fsdp, doc]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][fsdp, torch, liger-kernals] feat: dynamic batching`

### Test

> Both training scripts (`train_qwen25_omni.py`, `train_qwen3_omni.py`) now delegate to the same `process_sample_qwen_omni` entry point. Token IDs are resolved at runtime from the processor vocab instead of being hardcoded, so each model variant picks up the correct values automatically. Functional behavior is unchanged.

### API and Usage Example

> No public API change. The internal `process_sample` function in each omni training script is replaced by the shared `process_sample_qwen_omni` imported from `tasks.data.vlm_data_process`. Callers pass the same arguments as before.

```python
from tasks.data.vlm_data_process import process_sample_qwen_omni

transform = partial(
    process_sample_qwen_omni,
    processor=processor,
    position_id_func=position_id_func,
    **args.data.mm_configs,
)
```

### Design & Code Changes

> - **`tasks/data/vlm_data_process.py`**: Added `get_omni_token_ids`, private `_process_sample_omni`, and public `process_sample_qwen_omni`. Token IDs for image/video/audio are resolved from the tokenizer vocab at call time, with safe fallbacks to known defaults.
> - **`tasks/omni/train_qwen25_omni.py`**: Removed local `process_sample` (~135 lines), `SYSTEM_MESSAGE`, and now-unused imports. Replaced with a single import of `process_sample_qwen_omni`.
> - **`tasks/omni/train_qwen3_omni.py`**: Same cleanup as above (~136 lines removed).

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [ ] Read the [Contribute Guide](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md).
- [ ] Apply [pre-commit checks](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md?plain=1#L22): `make commit && make style && make quality`
- [ ] Add / Update [the documentation](https://github.com/ByteDance-Seed/VeOmni/blob/main/docs).
- [ ] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/ByteDance-Seed/VeOmni/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...

### CODE REVIEW

File: tasks/data/vlm_data_process.py [hunk 1]
```diff
@@ -159,3 +162,174 @@ def process_sample_qwen3_vl(
...
+def get_omni_token_ids(processor: "ProcessorMixin") -> tuple[int, int, int]:
+    """
+    Resolve (image_token_id, video_token_id, audio_token_id) by reading from the processor's
+    tokenizer vocab. Supports both Qwen2.5-Omni and Qwen3-Omni-MoE:
+      Qwen2.5-Omni:   image=151655 (<|IMAGE|>),     video=151656 (<|VIDEO|>),     audio=151646 (<|AUDIO|>)
+      Qwen3-Omni-MoE: image=151655 (<|image_pad|>), video=151656 (<|video_pad|>), audio=151675 (<|audio_pad|>)
+    """
+    tokenizer = getattr(processor, "tokenizer", processor)
+    vocab = tokenizer.get_vocab()
+    # Qwen2.5-Omni uses <|IMAGE|>/<|VIDEO|>/<|AUDIO|>; Qwen3-Omni uses <|image_pad|>/<|video_pad|>/<|audio_pad|>
```
Comment (piyifan123) [inline tasks/data/vlm_data_process.py]: nit: link tokenizer and config json in huggingface so that it's more obvious to readers.

File: tasks/data/vlm_data_process.py [hunk 2]
```diff
@@ -159,3 +162,174 @@ def process_sample_qwen3_vl(
...
+    image_token_id = vocab.get("<|image_pad|>", vocab.get("<|IMAGE|>", 151655))
```
Comment (piyifan123) [inline tasks/data/vlm_data_process.py]: nit: not to fallback to a magic number to avoid unexpected behavior.

File: tasks/data/vlm_data_process.py [hunk 3]
```diff
@@ -159,3 +162,174 @@ def process_sample_qwen3_vl(
...
+    else:
+        audio_audios = []
+
+    audios = []
+    for item in input_conversations:
+        for content in item["content"]:
+            if content["type"] == "video":
+                audios.append(video_audios.pop(0))
+            elif content["type"] == "audio":
+                audios.append(audio_audios.pop(0))
```
[the remark this answers is not in the document]
Comment (piyifan123) [reply tasks/data/vlm_data_process.py]: can we check this?

File: tasks/data/vlm_data_process.py [hunk 4]
```diff
@@ -159,3 +162,174 @@ def process_sample_qwen3_vl(
...
+    model_inputs["image_mask"] = image_mask
+    model_inputs["video_mask"] = video_mask
+    model_inputs["audio_mask"] = audio_mask
+    input_ids[image_mask | video_mask | audio_mask] = 0
+    model_inputs["input_ids"] = input_ids
+    model_inputs["attention_mask"] = model_inputs["attention_mask"].squeeze(0)
+
+    labels = torch.full_like(input_ids, fill_value=IGNORE_INDEX)
+    user_start_index = torch.where(input_ids == 872)[0].tolist()  # "user" 872
+    assistant_start_index = torch.where(input_ids == 77091)[0].tolist()  # "assistant" 77091
```
[the remark this answers is not in the document]
Comment (piyifan123) [reply tasks/data/vlm_data_process.py]: please check

File: tasks/data/vlm_data_process.py (hunk 1 above)
Comment (TimYangst) [reply tasks/data/vlm_data_process.py]: Done.

File: tasks/data/vlm_data_process.py (hunk 2 above)
Comment (TimYangst) [reply tasks/data/vlm_data_process.py]: Done

File: tasks/data/vlm_data_process.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (TimYangst) [reply tasks/data/vlm_data_process.py]: Done

File: tasks/data/vlm_data_process.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (TimYangst) [reply tasks/data/vlm_data_process.py]: Done
